extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/QuarkusRestPathTemplateInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.resteasy.runtime;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import io.quarkus.arc.ArcInvocationContext;
+import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
+import io.vertx.core.http.impl.HttpServerRequestInternal;
+
+@SuppressWarnings("unused")
+@QuarkusRestPathTemplate
+@Interceptor
+@Priority(Interceptor.Priority.LIBRARY_BEFORE + 10)
+public class QuarkusRestPathTemplateInterceptor {
+    @Inject
+    CurrentVertxRequest request;
+
+    @AroundInvoke
+    Object restMethodInvoke(InvocationContext context) throws Exception {
+        QuarkusRestPathTemplate annotation = getAnnotation(context);
+        try {
+            return context.proceed();
+        } finally {
+            if (annotation != null) {
+                ((HttpServerRequestInternal) request.getCurrent().request()).context().putLocal("UrlPathTemplate",
+                        annotation.value());
```

Comment:
Note that this won't deal with subresources, but I don't know how big an issue this is. As sub resources are directly instantiated by the user code you can't guarantee that interceptors will be applied.

Reply:
would a subresource have a path annotation?

Reply:
Well the initial resource locator method will, however the resource that is returned likely won't be managed by CDI so won't be intercepted.